### DISCUSSION TITLE
DOSbox-X support

### DISCUSSION THREAD

**Discussion by Torinde** [640-KB/GLaBIOS#49] [Suggestions]
DOSbox-X supports [loading custom BIOS](https://github.com/joncampbell123/dosbox-x/issues/193). I tried with IBM PC BIOS, but couldn't find settings combination that works with it.

Would it possible to make GLaBIOS run there?

**Comment by 640-KB** [comment]
Hi there.  So we actually experimented with this a while ago and were able to get it to go through the POST test ([image](https://cdn.discordapp.com/attachments/1012479517253709965/1192560804281733170/image.png?ex=6672928c&is=6671410c&hm=cf31f05b103ef0ce15963a1497fa428fd5888c2456745087197e73fc6b6c2c93)), but I don't believe were ever successful in getting a DOSBox disk driver to work natively - either a floppy disk or a hard drive via XUB.  This [question has asked here](https://github.com/joncampbell123/dosbox-x/issues/4735), but I don't believe it was ever fully resolved.  If you are on the DOSBox-X Discord, there was some more [discussion about this too here](https://discord.com/channels/449249801197060096/485564093881647125/1191848512346931301).

I recall it took some fiddling, but using the  `boot -bios` command did work to boot it, though DOSBox-X doesn't implement the XT keyboard interrupt/PPI, so the keyboard POST tests failed.  We made [this version](https://cdn.discordapp.com/attachments/1012479517253709965/1191824610182840461/GLABIOS_0.2.6_8E5.ROM?ex=667287ea&is=6671366a&hm=fadf2ff00c4d8f702b48cba1f10874feadf6c9ecf4730f4367eb711a7f3079b5&) that just removed those tests and got it to at least attempt to boot.  However, weren't able to progress past that because DOSBox-X (apparently) doesn't completely implement the legacy floppy drive controller, and also (if I recall correctly) the DOSBox-X implementation of IDE uses the standard 16 bit high-IRQs (14 and 15) and we weren't successful in creating a build of XUB that would work that way.

Also, I believe that we had to use CGA or MDA since the VGA in DOSBox-X did not seem to implement a video BIOS as an option ROM (again, iirc). 

Maybe try using CGA and also that GLaBIOS ROM above and see if you at least get as far as we did?  Would definitely be great to see if we can get it to boot someday.

640KB

**Reply by Torinde** [reply]
Thanks! Can you please post the files here?

**Reply by 640-KB** [reply]
Oh strange, those links have stopped working.  Here they are.  

POST test image
![POST test image](https://github.com/640-KB/GLaBIOS/assets/23486433/dc19e761-f424-4360-a735-8be837de9ba9)

GLABIOS Test image: [GLABIOS_0.2.6_8E5.zip](https://github.com/user-attachments/files/15904349/GLABIOS_0.2.6_8E5.zip). The only thing we really did here was disable the keyboard controller tests so that it didn't give you an error on boot (but nothing that was really specific to DOSBox-X). We just got stuck trying to find a way to boot from DOSBox-X's hard drive controller emulation, which is AT compatible. I believe XT Universal BIOS can be built in such a way to do it, but we just didn't get that far.

I do recall I got that "Illegal Callback #27 Called" error that [others have reported](https://github.com/joncampbell123/dosbox-x/issues/193#issuecomment-1837178214) and now I can't recall exactly what I did to finally get it to work. Maybe it was re-install DOSBox-X, or fiddle with the video drivers.  I believe it does have to be CGA, if I recall correctly.

Let me know if you have any success and maybe we can make a wiki page about it!

**Reply by Torinde** [reply]
I'll try... maybe without HDD, only FDD - there are some .conf settings specifically for XT
